File: src/main/java/com/redbox/domain/attach/strategy/NoticeFileStrategy.java
package com.redbox.domain.attach.strategy;

import com.redbox.domain.attach.entity.AttachFile;
import com.redbox.domain.attach.entity.Category;
import com.redbox.domain.notice.entity.Notice;
import com.redbox.domain.notice.exception.NoticeNotFoundException;
import com.redbox.domain.notice.repository.NoticeRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class NoticeFileStrategy implements FileAttachStrategy {

    private final NoticeRepository noticeRepository;

    @Override
    public AttachFile attach(Long postId, String originalFilename, String newFilename) {
        Notice notice = noticeRepository.findById(postId)
                .orElseThrow(NoticeNotFoundException::new);

        return AttachFile.builder()
                .category(Category.NOTICE)
                .notice(notice)
                .originalFilename(originalFilename)
                .newFilename(newFilename)
                .build();
    }
}

File: migration/src/main/kotlin/com/redbox/domain/community/attach/strategy/NoticeFileStrategy.kt
package com.redbox.domain.community.attach.strategy

import com.redbox.domain.community.attach.entity.AttachFile
import com.redbox.domain.community.attach.entity.Category
import com.redbox.domain.community.notice.entity.Notice
import com.redbox.domain.community.notice.exception.NoticeNotFoundException
import com.redbox.domain.community.notice.repository.NoticeRepository
import org.springframework.stereotype.Component

@Component
class NoticeFileStrategy(
    private val noticeRepository: NoticeRepository
) : FileAttachStrategy {

    override fun attach(postId: Long?, originalFilename: String?, newFilename: String?): AttachFile {
        val id = requireNotNull(postId) { "postId cannot be null" }
        val notice: Notice = noticeRepository.findById(id)
            .orElseThrow { NoticeNotFoundException() }

        return AttachFile(
            category = Category.NOTICE,
            notice = notice,
            originalFilename = requireNotNull(originalFilename) { "originalFilename cannot be null" },
            newFilename = requireNotNull(newFilename) { "newFilename cannot be null" }

        )
    }
}
